avocado/utils/astring.py

```diff
@@ -35,6 +36,9 @@
 from six.moves import xrange as range
 
 
+#: Preferred encoding based on system locale (could be used as default)
+#: but in some cases might be wrong. Use it wisely!
+ENCODING = locale.getpreferredencoding()
```

Comment:
On API documentation isn't clear if getpreferredencoding() may return None. If so you may want to convert it to a function that returns 'utf-8' in case of None.

Also shouldn't you pass False to getpreferredencoding() then it doesn't call setlocale()?

Reply:
I don't really have a good answer to this. When I try to break it, it reports 'ANSI_X3.4-1968' (ASCII). I think we can refine this later (btw this is another reason for having standardize way and not just relying on people remembering to call the proper `locale` function).

As for the setlocale I also have no clue. Maybe it is needed, maybe it is not. I chose the default to start with something and when we find issues we should address them in this single place.

Reply:
The "docstring" that will make into the documentation has to be clearer.  It must say that it represents a dynamic value, that is system/environment specific, and that the value shown just happens to be the value on the system that generated the documentation.  Also, it'd be helpful to put a link do the Python docs on the `locale.getpreferredencoding()` function, which is really the essence of this variable.

Reply:
OK